pkgs/development/libraries/libblocksruntime/default.nix

```diff
@@ -0,0 +1,27 @@
+{ stdenv, fetchFromGitHub, clang }:
+
+stdenv.mkDerivation {
+  name = "blocksruntime";
+
+  src = fetchFromGitHub {
+    owner = "mackyle";
+    repo = "blocksruntime";
+    rev = "b5c5274daf1e0e46ecc9ad8f6f69889bce0a0a5d";
+    sha256 = "0ic4lagagkylcvwgf10mg0s1i57h4i25ds2fzvms22xj4zwzk1sd";
+  };
+
+  buildInputs = [ clang ];
+
+  configurePhase = ''
+    export CC=clang
```

Comment:
Hmm.  So it's a bit tricky I think, here's what I know/why things are why this way:
* libblocksruntime isn't C++ (so I don't think it matters if we use libc++)
* their build script checks for gcc, clang, then 'cc' but later uses the name it found to determine what CFLAGS to use.  Using a clang-based stdenv doesn't actually make clang available by that name (just like linux stdenv doesn't make gcc available by that name), so clang would need to be a buildInput or this behavior patched around.
* It turns out that hardening flags add -O2 which I think may make the previous point moot but I don't think that should be relied upon.

I'm not saying this is the best solution, but I'm not sure it's better to use clangStdenv.  Thoughts? Incidentally I tried using clangStdenv just now and the resulting archive was bit-identical but that doesn't really mean much just a fun fact :).